<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DEV_CER</class>
 <widget class="QMainWindow" name="DEV_CER">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>450</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Cifrados</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true">
    background-color: rgb(240, 248, 255);
    </string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QFrame" name="headerFrame">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayoutHeader">
       <item>
        <widget class="QLabel" name="titleLabel">
         <property name="styleSheet">
          <string notr="true">
          color: rgb(25, 42, 86);
          font: 87 16pt "Arial Black";
          </string>
         </property>
         <property name="text">
          <string>Simulador de Cifrados</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="contentFrame">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayoutContent">
       <item>
        <widget class="QFrame" name="inputFrame">
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QVBoxLayout" name="verticalLayoutInput">
          <item>
           <widget class="QLabel" name="wordLabel">
            <property name="text">
             <string>TEXTO ORIGINAL</string>
            </property>
            <property name="styleSheet">
             <string notr="true">color: rgb(0, 102, 204);</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTextEdit" name="textToEncrypt">
            <property name="styleSheet">
             <string notr="true">
             background-color: rgb(224, 242, 255);
             border: 1px solid rgb(0, 102, 204);
             </string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayoutCipherOptions">
            <item>
             <widget class="QLabel" name="cipherTypeLabel">
              <property name="text">
               <string>Tipo de Cifrado:</string>
              </property>
              <property name="styleSheet">
               <string notr="true">color: rgb(0, 102, 204);</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="cipherType">
              <property name="styleSheet">
               <string notr="true">
               background-color: rgb(224, 242, 255);
               border: 1px solid rgb(0, 102, 204);
               </string>
              </property>
              <item>
               <property name="text">
                <string>Cifrado César con palabra clave</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Cifrado César con secuencia</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Cifrado Vigenère</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="keyLabel">
            <property name="text">
             <string>Clave:</string>
            </property>
            <property name="styleSheet">
             <string notr="true">color: rgb(0, 102, 204);</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="keyInput">
            <property name="styleSheet">
             <string notr="true">
             background-color: rgb(224, 242, 255);
             border: 1px solid rgb(0, 102, 204);
             </string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="encryptButton">
            <property name="text">
             <string>CIFRAR</string>
            </property>
            <property name="styleSheet">
             <string notr="true">
             background-color: rgb(30, 144, 255);
             color: white;
             </string>
            </property>
           </widget>
          </item>
          <!-- Botón de limpiar texto original -->
          <item>
           <widget class="QPushButton" name="clearTextToEncryptButton">
            <property name="text">
             <string>LIMPIAR TEXTO ORIGINAL</string>
            </property>
            <property name="styleSheet">
             <string notr="true">
             background-color: rgb(255, 165, 0);
             color: white;
             </string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="outputFrame">
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QVBoxLayout" name="verticalLayoutOutput">
          <item>
           <widget class="QLabel" name="outputLabel">
            <property name="text">
             <string>TEXTO CIFRADO</string>
            </property>
            <property name="styleSheet">
             <string notr="true">color: rgb(0, 102, 204);</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTextEdit" name="decryptedText">
            <property name="styleSheet">
             <string notr="true">
             background-color: rgb(224, 242, 255);
             border: 1px solid rgb(0, 102, 204);
             </string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="decryptButton">
            <property name="text">
             <string>DESCIFRAR</string>
            </property>
            <property name="styleSheet">
             <string notr="true">
             background-color: rgb(255, 69, 0);
             color: white;
             </string>
            </property>
           </widget>
          </item>
          <!-- Botón de limpiar texto cifrado -->
          <item>
           <widget class="QPushButton" name="clearDecryptedTextButton">
            <property name="text">
             <string>LIMPIAR TEXTO CIFRADO</string>
            </property>
            <property name="styleSheet">
             <string notr="true">
             background-color: rgb(255, 165, 0);
             color: white;
             </string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
